packages/mobile/src/tokens/saga.ts

```diff
@@ -266,44 +270,69 @@ export async function getERC20TokenContract(tokenAddress: string) {
   return new kit.web3.eth.Contract(erc20.abi, tokenAddress)
 }
 
-export async function getERC20TokenBalance(token: StoredTokenBalance, address: string) {
-  let balance = null
-  try {
-    const contract = await getERC20TokenContract(token.address)
-    balance = await contract.methods.balanceOf(address).call()
-  } catch (error) {
-    Logger.error(TAG, `error fetching balance for ${token.name}`, error)
-  }
-  return balance
+interface FetchedTokenBalance {
+  tokenAddress: string
+  balance: string
 }
 
-export function* fetchReadableTokenBalance(address: string, token: StoredTokenBalance) {
-  const balance: number | null = yield call(getERC20TokenBalance, token, address)
-  return {
-    ...token,
-    balance:
-      balance || balance === 0
-        ? new BigNumber(balance)
-            .dividedBy(new BigNumber(10).exponentiatedBy(token.decimals))
-            .toString()
-        : null,
+interface UserBalancesResponse {
+  userBalances: {
+    balances: FetchedTokenBalance[]
   }
 }
 
+export async function fetchTokenBalancesForAddress(
+  address: string
+): Promise<FetchedTokenBalance[]> {
+  const response = await apolloClient.query<UserBalancesResponse, { address: string }>({
+    query: gql`
+      query FetchUserBalances($address: Address!) {
+        userBalances(address: $address) {
+          balances {
+            tokenAddress
+            balance
+          }
+        }
+      }
+    `,
+    variables: {
+      address,
+    },
+    fetchPolicy: 'network-only',
+    errorPolicy: 'all',
+  })
+
+  return response.data.userBalances.balances
+}
+
 export function* importTokenInfo() {
-  // In e2e environment we use a static token list since we can't access Firebase.
-  const tokens: StoredTokenBalances = isE2EEnv
-    ? e2eTokens()
-    : yield call(readOnceFromFirebase, 'tokensInfo')
-  const address: string = yield select(walletAddressSelector)
-  const fetchedTokenBalances: StoredTokenBalance[] = yield all(
-    Object.values(tokens).map((token) => call(fetchReadableTokenBalance, address, token!))
-  )
-  const balances: StoredTokenBalances = {}
-  for (const tokenBalance of fetchedTokenBalances) {
-    balances[tokenBalance.address] = tokenBalance
+  try {
+    // In e2e environment we use a static token list since we can't access Firebase.
+    const tokens: StoredTokenBalances = isE2EEnv
+      ? e2eTokens()
+      : yield call(readOnceFromFirebase, 'tokensInfo')
+    const address: string = yield select(walletAddressSelector)
+    const tokenBalances: FetchedTokenBalance[] = yield call(fetchTokenBalancesForAddress, address)
```

Comment:
Can we add new analytics events for this?
We were discussing this with @grvalora for #1085
I'm now thinking we should add a new event name for this new multi token context, maybe `fetch_token_balances`, `fetch_token_balances_error`?
Can be done in a separate PR though. But is important to ship along with this feature otherwise gonna miss some data.
Though of course #1409 may be enough.

Well, let's sync to make sure we're on the same page?

Reply:
I added a `fetch_balance` event and verified that the updated balances are included per the changes on #1409